File: packages/core/src/types/events.ts
export interface KeystrokeEvent {
  event_type: string;
  key: string;
  code: string;
  timestamp: number;
  shift_key: boolean;
  ctrl_key: boolean;
  alt_key: boolean;
  meta_key: boolean;
  repeat: boolean;
  is_composing: boolean;
  sequence_number: number;
}

File: apps/desktop/src-tauri/src/capture/events.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct KeystrokeEvent {
    pub event_type: String,
    pub key: String,
    pub code: String,
    pub timestamp: f64,
    pub shift_key: bool,
    pub ctrl_key: bool,
    pub alt_key: bool,
    pub meta_key: bool,
    pub repeat: bool,
    pub is_composing: bool,
    pub sequence_number: u64,
}
